lib/note-content-editor.tsx

```diff
@@ -425,6 +426,44 @@ class NoteContentEditor extends Component<Props> {
     this.editor = editor;
     this.monaco = monaco;
 
+    monaco.languages.registerLinkProvider('plaintext', {
+      provideLinks: (model) => {
+        const matches = model.findMatches(
+          'simplenote://note/[a-zA-Z0-9-]+',
+          true, // searchOnlyEditableRange
+          true, // isRegex
+          false, // matchCase
+          null, // wordSeparators
+          false // captureMatches
+        );
+        const links = [];
+        matches.forEach(function (match) {
+          links.push({
+            range: match.range,
+            // don't set URL here because then Monaco skips resolveLink
+            // @cite: https://github.com/Microsoft/vscode/blob/8f89095aa6097f6e0014f2d459ef37820983ae55/src/vs/editor/contrib/links/getLinks.ts#L43:L65
+          });
+        });
+        return {
+          links: links,
+        };
+      },
+      resolveLink: (link) => {
+        const href = editor.getModel()?.getValueInRange(link.range);
+        if (!href) {
+          return;
+        }
+        const match = /^simplenote:\/\/note\/(.+)$/.exec(href);
+        if (!match) {
+          return;
+        }
+
+        const [fullMatch, linkedNoteId] = match;
+        this.props.openNote(linkedNoteId as T.EntityId);
+        link.url = '#'; // tell Monaco to do nothing and not complain about it
```

Comment:
let's avoid mutating the object passed-in to this function…

```ts
this.props.openNote(…);

return { ...link, url: '#' };
```

that creates a new copy instead of the "spooky action at a distance" involved when we change a property of the object passed in as an argument